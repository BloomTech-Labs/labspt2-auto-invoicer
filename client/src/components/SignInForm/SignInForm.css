.signin-form {
  width: 90%;
  min-height: 200px;
  margin: 10px auto;
  background: lightgray;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
}

.signin-form-group {
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 5px 2px;
  margin: 10px 0;
  min-width: 400px;
}

.signin-input {
  font-size: 2rem;
  padding: 5px 2px;
  border: 1px solid black;
  border-radius: 5px;
  margin: 0 10px;
}

.signin-input:hover {
  border: 1px solid gray;
}
