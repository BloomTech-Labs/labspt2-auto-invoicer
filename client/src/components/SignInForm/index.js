import React from "react";

// import icons
import EmailIcon from "@material-ui/icons/Email";
import PasswordIcon from "@material-ui/icons/LockOpen";
// import css here
import "./SignInForm.css";

class SignInForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      email: "",
      password: ""
    };
  }
  changeHandler = e => {
    return this.setState({ [e.target.name]: e.target.value });
  };
  createLoginObject = e => {
    e.preventDefault();
    const { email, password } = this.state;
    const userObject = { email, password };
    /* console.log until we have backend hooked up */
    console.log(userObject);
  };
  render() {
    return (
      <form className="signin-form" onSubmit={this.createLoginObject}>
        <label htmlFor="email" className="signin-form-group">
          <EmailIcon />
          <input
            className="signin-input"
            type="text"
            placeholder="email"
            name="email"
            id="email"
            onChange={this.changeHandler}
            value={this.state.email}
          />
        </label>
        <label htmlFor="password" className="signin-form-group">
          <PasswordIcon />
          <input
            className="signin-input"
            type="password"
            placeholder="password"
            name="password"
            id="password"
            onChange={this.changeHandler}
            value={this.state.password}
          />
        </label>
        <button className="authentication-btns" type="submit">
          Login
        </button>
      </form>
    );
  }
}

export default SignInForm;
