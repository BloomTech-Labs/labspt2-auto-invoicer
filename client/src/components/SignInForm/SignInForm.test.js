import React from "react";
import { render } from "react-testing-library";
import "jest-dom/extend-expect";
import SignInForm from "../SignInForm";

describe("<SignInForm />", () => {
  it("renders Sign In Form Component", () => {
    const { asFragment } = render(<SignInForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});
