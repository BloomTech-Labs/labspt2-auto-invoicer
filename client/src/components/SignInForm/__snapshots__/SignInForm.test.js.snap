// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInForm /> renders Sign In Form Component 1`] = `
<DocumentFragment>
  <form
    class="signin-form"
  >
    <label
      class="signin-form-group"
      for="email"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-1"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
      <input
        class="signin-input"
        id="email"
        name="email"
        placeholder="email"
        type="text"
        value=""
      />
    </label>
    <label
      class="signin-form-group"
      for="password"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-1"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 17c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm6-9h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6h1.9c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm0 12H6V10h12v10z"
        />
      </svg>
      <input
        class="signin-input"
        id="password"
        name="password"
        placeholder="password"
        type="password"
        value=""
      />
    </label>
    <button
      class="authentication-btns"
      type="submit"
    >
      Login
    </button>
  </form>
</DocumentFragment>
`;
